examples/fuzz_example1/trdelnik-tests/src/fuzz_instructions.rs

```diff
@@ -280,18 +301,28 @@ pub mod fuzz_example1_fuzz_instructions {
         fn check(
             &self,
             pre_ix: Self::IxSnapshot,
-            _post_ix: Self::IxSnapshot,
-            _ix_data: Self::IxData,
+            post_ix: Self::IxSnapshot,
+            ix_data: Self::IxData,
         ) -> Result<(), &'static str> {
-            if let Some(state) = pre_ix.state {
-                if !state.registrations_round {
-                    return Err("Registration instruction passed before registration activation!");
+            // INFO
+            // This fuzz check will reveal that invest can be performed
+            // even though registration windows was not closed.
+            if let Some(project_pre) = pre_ix.project {
+                let project_post = post_ix.project.unwrap();
+
+                if let Some(state) = pre_ix.state {
+                    if !state.registrations_round
```

Comment:
I don't understand this check. To me it looks like it only checks the regular condition, when it is possible to invest (registration_round == false). To detect a problem, we would either need to check the value of registration_round after initialize ix or we would need to check, if the end_registrations ix was invoked before the invest ix - which is currently not possible. At first I thought it is not necessary to have access to the instruction sequence from the check method (because a program is stateless anyway), but it might be useful as this example shows.